### PR TITLE
Add Snyk plugin

### PR DESCRIPTION
**Pull request by schultzp2020**
## Description

Please explain the changes you made here.

- Adds the Snyk plugin

## Which issue(s) does this PR fix

- Fixes https://github.com/janus-idp/backstage-showcase/issues/61

## PR acceptance criteria

_Put an `x` in the boxes that apply_

- [x] GitHub Actions are completed and successful
- [x] Unit Tests are updated and passing
- [x] E2E Tests are updated and passing
- [x] Documentation is updated if necessary

## How to test changes / Special notes to the reviewer

N/A

### CODE REVIEW

File: app-config.yaml [hunk 1]
```diff
@@ -56,6 +56,11 @@ proxy:
       # Authorization: 'Bearer <YOUR TOKEN>'
     changeOrigin: true
     secure: true
+  '/snyk':
+    target: https://snyk.io/api/v1
+    headers:
+      User-Agent: tech-services/backstage-plugin/1.0
+      Authorization: token ${SNYK_TOKEN}
```
Comment (PatAKnight) [inline app-config.yaml:63]: Is there any way you could add this information to the [getting-started documentation](https://github.com/janus-idp/backstage-showcase/blob/main/docs/getting-started.md). We have other teams interested in using our image and having the configurations present in the doc will help them get setup.

File: catalog-entities/components.yaml [hunk 2]
```diff
@@ -8,13 +8,15 @@ metadata:
...
   annotations:
-    argocd/app-name: "janus-idp-smaug"
-    backstage.io/kubernetes-id: "janus-idp"
+    argocd/app-name: janus-idp-smaug
+    backstage.io/kubernetes-id: janus-idp
     github.com/project-slug: janus-idp/backstage-showcase
     quay.io/repository-slug: janus-idp/backstage-showcase
     backstage.io/kubernetes-namespace: janus-idp
+    snyk.io/org-name: janus-idp
+    snyk.io/project-ids: ${SNYK_PROJECT_IDS}
```
Comment (PatAKnight) [inline catalog-entities/components.yaml:19]: Can you also add this to the [catalog-info](https://github.com/janus-idp/backstage-showcase/blob/main/catalog-info.yaml).

**Comment by christophe-f** [comment]
Did you create a new issue in upstream related to the 404 error. Can add the link to #61

**Comment by christophe-f** [comment]
Closing this PR as we can't use the Snyk APIs access with our project and test this plugin. We can re-open once Snyk is giving an open source access.

**Pull request closed without merging**